caffeine@patapon.info/extension.js

```diff
@@ -183,27 +182,34 @@ class CaffeineToggle extends QuickSettings.QuickMenuToggle {
     _syncTimers(resetDefault) {
         this._itemsSection.removeAll();
         this._timerItems.clear();
-        const durationIndex = this._settings.get_int(DURATION_TIMER_INDEX);
+        // Get duration list and add '0' for the 'infinite' entry (no timer)
+        let durationValues = this._settings.get_value(DURATION_TIMER_LIST).deepUnpack();
+        durationValues.push(0);
 
-        for (const timer of TIMERS) {
+        // Create menu timer
+        for (const [index, timer] of durationValues.entries()) {
             let label = null;
-            if (timer[0] === 0) {
+            if (timer === 0) {
                 label = _('Infinite');
             } else {
-                label = parseInt(timer[durationIndex]) + _(' minutes');
+                let hours = Math.floor(timer / 3600);
+                let minutes = Math.floor((timer % 3600) / 60);
+                if (parseInt(hours) !== 0) {
```

Comment:
If `hours` is already `Math.floor(...)`, why is `parseInt()` called on it? `hours` will already be an integer at this point, unless I misunderstood

Same for `minutes`

Reply:
Absolutely, I'll fix it.